plutus-core/plutus-core/src/PlutusCore/Error.hs

```diff
@@ -111,8 +120,30 @@ data Error uni fun ann
 makeClassyPrisms ''Error
 deriving stock instance (Show fun, Show ann, Closed uni, Everywhere uni Show, GShow uni, Show ParserError) => Show (Error uni fun ann)
 
-instance AsParserError (Error uni fun ann) where
-    _ParserError = _ParseErrorE
+-- To be able to carry megaparsec's @ParseErrorBundle@ straight instead of the wrapper @ParserErrorBundle@:
+-- Construct a simple prism of Prism' (Error uni fun ann) ParserErrorBundle with
+-- prism' :: (a -> s) -> (s -> Maybe a) -> Prism' s a
+instance AsParserErrorBundle (Error uni fun ann) where
+    _ParserErrorBundle = prism' putError takeError
+
+-- (a -> s) function that specifies how we put a @ParserErrorBundle@ into @Error uni fun ann@.
+putError :: ParserErrorBundle -> Error uni fun ann
+putError (ParseErrorB a) = ParseErrorE a
+
+-- (s -> Maybe a) function that specifies how we get a @ParserErrorBundle@ from a @Error uni fun ann@.
+takeError :: Error uni fun ann -> Maybe ParserErrorBundle
+takeError (ParseErrorE a) = Just (ParseErrorB a)
+takeError _               = Nothing
+
+-- this instance is needed for @goldenPirM@.
```

Comment:
That seems suspicious to me. Perhaps we need to tweak `goldenPir`.

Reply:
I think it should be easy to fix? Something is just expecting the wrong kind of error.

Reply:
I think I can see the problem: in the tests you're repeatedly fixing the error type to `ParseErrorBundle T.Text ParserError` when you should be fixing it to `ParserErrorBundle`